templates/puppet.spec

```diff
@@ -0,0 +1,52 @@
+%{!?upstream_version: %global upstream_version %{version}%{?milestone}}
+{% if metadata.from_puppetlabs -%}
+%%define upstream_name {{ metadata.puppetlabs_name }}
+{% endif %}
+Name:                   {{ metadata.name }}
+Version:                XXX
+Release:                XXX
+Summary:                {{ metadata.summary }}
+License:                {{ metadata.license }}
+
+URL:                    {{ metadata.project_page }}
+
+Source0:                {{ metadata.source0 }}
+
+BuildArch:              noarch
+
+{% for dep in metadata.dependencies -%}
+Requires:               {{ dep.name }}
+{% endfor %}
+Requires:               puppet >= 2.7.0
+
+%description
+{{ metadata.description }}
+
+%prep
+%setup -q -n {{ '%{'+metadata.upstream_name+'}-%{upstream_version}' }}
```

Comment:
Then are all our current specs broken?  I see them using %{name} here, and this template keeps that behavior.  If we need to support this as well, I can add it.